build-support/bin/generate_lockfile.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+requirements_file=3rdparty/python/requirements.txt
+constraints_file=3rdparty/python/constraints.txt
+
+VENV=$(mktemp -d -t pants_lockfile_venv.XXXXX)
+virtualenv "${VENV}"
+"${VENV}/bin/pip" install -r "${requirements_file}"
```

Comment:
Agreed with Eric here, its cheap to replace `"${requirements_file}"`  with `"$(./pants dependencies --type=3rdparty ::)"` and keep this right from the get go.

Reply:
> Consider the algorithm described in https://www.pantsbuild.org/docs/python-third-party-dependencies#tip-set-up-a-virtual-environment-optional so that we can handle inline requirements.

As an aside, `./pants dependencies` only shows reqs that we actually depend on somewhere. For example, turns out we don't depend on beautifulsoup any more, but it's still in our requirements.txt. Probably the right thing to do in our case is just remove it from requirements.txt. But it does mean that the algorithm we describe there may not be quite right for users. But no need to worry about that just yet.

Reply:
Done (had to use tags to ignore a couple of test reqs).

Reply:
And the beautifulsoup thing was a lie, we do depend on it, but in build-support/bin. And `./pants dependencies` isn't picking up on that for some reason.

Reply:
OK, fixed that.